modules/fundamental/src/sbom/model/mod.rs

```diff
@@ -36,11 +45,24 @@ pub struct SbomPackage {
     pub name: String,
     pub version: Option<String>,
     #[serde(default, skip_serializing_if = "Vec::is_empty")]
-    pub purl: Vec<String>,
+    pub purl: Vec<SbomPackagePurl>,
     #[serde(default, skip_serializing_if = "Vec::is_empty")]
     pub cpe: Vec<String>,
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq, ToSchema)]
+#[serde(untagged)]
+pub enum SbomPackagePurl {
```

Comment:
I am not sure I am happy with that. I think it would be ok if that would be an input type (allowing more flexible input always seems good). But to my understanding, that's an output type. And I think in that case it would make sense for the caller to understand what can be expected as a return type.

Having a struct returning both purl and details would be fine I guess. But having either this or that, in the same array, might be quite complex.

Reply:
Sorted the coalesce, removed `SbomPackagePurl` enum...